pilot/pkg/ambient/controller/workloadcache.go

```diff
@@ -77,18 +95,29 @@ func (wc *workloadCache) Reconcile(key types.NamespacedName) error {
 
 	w := ambientpod.WorkloadFromPod(pod)
 	index, ok := wc.indexes[pod.Labels[ambient.LabelType]]
+	update := true
 	if ok && wc.validate(w) {
+		cur := index.ByNamespacedName[key]
+		if cur.Equals(w) {
+			update = false
+		}
 		// known type, cache it
 		index.Insert(w)
 	} else {
 		// if this Pod went from valid -> empty/invalid we need to remove it from every index
 		wc.removeFromAll(key)
 	}
-	wc.xds.ConfigUpdate(&model.PushRequest{
-		// TODO scope our updates
-		Full:   true,
-		Reason: []model.TriggerReason{model.AmbientUpdate},
-	})
+	if update {
```

Comment:
Why only for update? 

Should we also trigger for transition from `valid -> invalid`?